### PR TITLE
new lesson inheritance and constructors

### PR DESCRIPTION
**Pull request by linda-seiter**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,31 +2,630 @@
...
+Creating Teacher
+Teacher state:name=Tal, age=40
+Creating Student
+Student state:name=Hao, age=24, favoriteSubject=Java programming
+```
+
+## Default No-Args Constructors
+
+Recall that Java generates a no-args constructor if a class does not explicitly
+define a constructor.  Let's update each class to add a
```
Comment (kcleland0818) [inline README.md]: "...add a explicit no-args constructor..."

Change "a" to "an"

File: README.md [hunk 2]
```diff
@@ -2,31 +2,630 @@
...
+    }
+
+    // other methods
+
+}
+```
+
+Executing `Main.main()` results in the output:
+
+![constructor output](https://curriculum-content.s3.amazonaws.com/6677/pillars/constructor_output.png)
```
Comment (kcleland0818) [inline README.md:162]: Good job at showing this!

File: README.md [hunk 3]
```diff
@@ -2,31 +2,630 @@
...
+        System.out.println("Person state:" + person); 
+
+        //Teacher instance, inherits fields and methods from Person
+        //...
+    }
+}
+```
+
+Running the code shows the `Person(String name, int age)` constructor
+executing for the call `new Person("Kai", 32);`.
```
Comment (kcleland0818) [inline README.md]: I don't think we need the word "for" in this sentence?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.example</groupId>
+    <artifactId>aws-inheritance-constructors-solution</artifactId>
```
Comment (kcleland0818) [inline pom.xml]: I think once this is merged in, it'll just be java-inheritance-constructors

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md]: fixed

File: pom.xml (hunk 4 above)
Comment (linda-seiter) [reply pom.xml]: fixed

**Review by kcleland0818 — APPROVED** [review]

**Review by developerjaym — APPROVED** [review]

**Review by kcleland0818 — APPROVED** [review]

**Review by AlveeM — APPROVED** [review]

**Pull request merged**